libraries/common/include/common/vrf_wrapper.hpp

```diff
@@ -54,7 +54,22 @@ class VrfSortitionBase {
   }
 
  private:
-  void thresholdFromOutput() const { threshold_ = (((uint16_t)output_[1] << 8) | output_[0]); }
+  void thresholdFromOutput(uint16_t stake) const {
+    threshold_ = (((uint16_t)output_[1] << 8) | output_[0]);
+    if (stake > 1) {
+      uint16_t min_threshold = threshold_;
+      uint16_t threshold_candidate = threshold_;
+      // Generate sequence of thresholds using simple generator with original threshold as seed
+      for (uint16_t stake_counter = 1; stake_counter < stake; stake_counter++) {
+        const uint32_t a = 48271;  // Multiplier term used by C++11's minstd_rand
```

Comment:
I think it shouldn't be inside of this for cycle. And why is it uint32_t result of its multiplication assigns to uint16_t? I think it should be also uint16_t

Reply:
yes, I will move it out and change to uint16_t